### DISCUSSION TITLE
Maui iOS & BLE Issue

### DISCUSSION THREAD

**Discussion by derek-willis** [shinyorg/shiny#1147] [Q&A]
We are trying to adopt Shiny as a way forward with mobile apps. I have built the Sample.Maui app and it ran ok on Android but I cannot deploy it on iOS, the error in Visual Studio is: -

 ApplicationVerificationFailed: Failed to verify code signature of /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.QHK7rS/extracted/Sample.Maui.app : 0xe8008014 (The executable contains an invalid signature.)

If I build a Maui app from your template with BLE Client ticked then I am able to deploy on Android and iOS. What can be the difference between Sample.Maui and your generated project deployment?

I really need to test Sample.Maui on iOS (BLE CLient) because of the following reason: -

I generate a Maui project from your Visual Studio template with BLE Client ticked. I then add just a small piece of code for BLE scanning to prove I can see devices. This builds and runs fine on Android but on iOS the Scan function causes a fatal crash (native error it seems).

We use Visual Studio 2022 Version 17.4.1

**Comment by aritchie** [comment]
Please ensure you are running the latest version of the template as it is being updated often.  I just ran a test with the latest version and iOS, Android, & maccatalyst all work fine with BLE.  

If you are still facing the issue, please file a bug report for the template here: https://github.com/shinyorg/templates
Please include a screenshot of what options you used.

Also note, Shiny for MAUI (v3) is alpha.  We aren't taking issues for it directly right now

**Reply by derek-willis** [reply]
Thanks for the quick reply. I just updated the templates and created a new project. The packages look in a bad state and I cannot build the project unfortunately. Below is the images of the template selections and my generated project structure.

![image](https://user-images.githubusercontent.com/103993555/205134786-1e31ab3a-b7a2-4c14-99b2-cf5f184ba05d.png)
![image](https://user-images.githubusercontent.com/103993555/205134876-97299bad-e171-4a71-b363-bbc5fc0afca3.png)
![image](https://user-images.githubusercontent.com/103993555/205134994-f75a8c3e-26f2-4830-ba2f-2d76e56062d7.png)

The reason why I cannot deploy the Sample.Maui app is still a big mistery. I need to do this to verify that BLE works on iOS.

**Reply by derek-willis** [reply]
If the templates produce projects based on v3 but you are not taking issues on v3 yet, how do I proceed ?

**Reply by aritchie** [reply]
BLE is working fine for me right now using the latest templates.  I've run a scan without issues.  The templates often take time to restore nuget packages.  You can try restarting VS as well.  Perhaps your VS setup is missing some components...?  I have no idea because both vs4mac & vs4win are working with this configuration right now on my machines.

Sample.Maui is mainly to show off code and used by me for testing.  I haven't worked with it much in the past few weeks so it may be broken because again... it is alpha, not a stable release.

> If the templates produce projects based on v3 but you are not taking issues on v3 yet

Templates either work or they don't.  They produce a consistent product that I can at least see what might be missing.  I went over the template setup with this configuration and everything looks correct.  The templates are marked as alpha as well, because Shiny v3 is alpha.  

> how do I proceed ?

You can wait until I send a stable release, contribute, etc.  It's OSS.  Do what works for you.  However, I don't see an issue on this end specifically for what you've shown.

**Reply by aritchie** [reply]
I just tried Sample.Maui on my device.  It is deploying and scan stuff is working fine.  

> ApplicationVerificationFailed: Failed to verify code signature of /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.QHK7rS/extracted/Sample.Maui.app : 0xe8008014 (The executable contains an invalid signature.)

This is caused because you're missing provisioning for either your device, the appid, or some combination.  You can use automatic provisioning and it will create the necessary provisioning if you have setup your apple account on visual studio

**Comment by derek-willis** [comment]
Its been tried with automatic and manual provisioning, even created new profiles fresh. The template generated projects will deploy fine. The exception is projects with todays updated templates as the projects wont even build.

On 1 December 2022 19:55:52 GMT, Allan Ritchie ***@***.***> wrote:
>I just tried Sample.Maui on my device.  It is deploying and scan stuff is working fine.  
>
>> ApplicationVerificationFailed: Failed to verify code signature of /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.QHK7rS/extracted/Sample.Maui.app : 0xe8008014 (The executable contains an invalid signature.)
>
>This is caused because you're missing provisioning for either your device, the appid, or some combination.  You can use automatic provisioning and it will create the necessary provisioning if you have setup your apple account on visual studio
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/orgs/shinyorg/discussions/1147#discussioncomment-4286553
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by derek-willis** [comment]
We would certainly contribute to the great work but i must prove to my leaders that ble works on androud and ios. All the issues are iOS. 

Can you tell me what version of xcode you use and mac version. 

I just feel there is something simple that needs adjusting somewhere. 

All i have done is get shiny from github and build it. We build maui apps all the time so are used to the environment.

Our Visual Studio is on Windows 11

A trick is being missed somewhere

On 1 December 2022 19:26:04 GMT, Allan Ritchie ***@***.***> wrote:
>BLE is working fine for me right now using the latest templates.  I've run a scan without issues.  The templates often take time to restore nuget packages.  You can try restarting VS as well.  Perhaps your VS setup is missing some components...?  I have no idea because both vs4mac & vs4win are working with this configuration right now on my machines.
>
>Sample.Maui is mainly to show off code and used by me for testing.  I haven't worked with it much in the past few weeks so it may be broken because again... it is alpha, not a stable release.
>
>> If the templates produce projects based on v3 but you are not taking issues on v3 yet
>
>Templates either work or they don't.  They produce a consistent product that I can at least see what might be missing.  I went over the template setup with this configuration and everything looks correct.  The templates are marked as alpha as well, because Shiny v3 is alpha.  
>
>> how do I proceed ?
>
>You can wait until I send a stable release, contribute, etc.  It's OSS.  Do what works for you.  However, I don't see an issue on this end specifically for what you've shown.
>
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/orgs/shinyorg/discussions/1147#discussioncomment-4286370
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Reply by aritchie** [reply]
Please don't treat this like technical support.  This isn't a paid product.

I'm telling you that the error with Sample.Maui within the Shiny repo that you linked in your first comment is from a provisioning problem.  That is fact - the build message says so.  I have a labelled appId in called org.shiny.samples which means your profile has to match that.  There is nothing more I can do to help you there. 

The template also isn't giving me any errors or build time exceptions.  What is the exception you are getting?

I am always using the latest version of visual studio and xcode

**Comment by derek-willis** [comment]
I think you misunderstand me. I certainly do not treat this as technical support. I am spending a lot of time trying to get it to work, it doesnt on iOS so some form of help is needed. I am trying hard to adopt your efforts. We would love to pay for this solutiion but, it must work out of the box and not require huge effort to get things working.

Your machine enviromnent may contain crucial artifacts that may not exist on someone elses machine, this is a very common issue. To say 'it works on my machine' is the sign that it has not been tested in a new and clean installation. 

I too are using the latest versions of VS and XCODE, on Windows 11.

The exception is when I call the scan function from a button click just the same as in the sample code. The exception is a non user friendly crash, a native memory dump is produced etc. I can capture images of this and send to you. Same project code for Android does not crash and i receive a scan response with ble devices in a list of peripherals. The crash occurs in debug and release compliations.

Hope this helps a little more.

On 1 December 2022 20:10:06 GMT, Allan Ritchie ***@***.***> wrote:
>Please don't treat this like technical support.  This isn't a paid product.
>
>I'm telling you that the error with Sample.Maui within the Shiny repo that you linked in your first comment is from a provisioning problem.  That is fact - the build message says so.  I have a labelled appId in called org.shiny.samples which means your profile has to match that.  There is nothing more I can do to help you there. 
>
>The template also isn't giving me any errors or build time exceptions.  What is the exception you are getting?
>
>I am always using the latest version of visual studio and xcode
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/orgs/shinyorg/discussions/1147#discussioncomment-4286636
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by derek-willis** [comment]
Clarification, xcode on a mac brick, vs on windows 11 on pc.

On 1 December 2022 22:35:06 GMT, Derek ***@***.***> wrote:
>I think you misunderstand me. I certainly do not treat this as technical support. I am spending a lot of time trying to get it to work, it doesnt on iOS so some form of help is needed. I am trying hard to adopt your efforts. We would love to pay for this solutiion but, it must work out of the box and not require huge effort to get things working.
>
>Your machine enviromnent may contain crucial artifacts that may not exist on someone elses machine, this is a very common issue. To say 'it works on my machine' is the sign that it has not been tested in a new and clean installation. 
>
>I too are using the latest versions of VS and XCODE, on Windows 11.
>
>The exception is when I call the scan function from a button click just the same as in the sample code. The exception is a non user friendly crash, a native memory dump is produced etc. I can capture images of this and send to you. Same project code for Android does not crash and i receive a scan response with ble devices in a list of peripherals. The crash occurs in debug and release compliations.
>
>Hope this helps a little more.
>
>On 1 December 2022 20:10:06 GMT, Allan Ritchie ***@***.***> wrote:
>>Please don't treat this like technical support.  This isn't a paid product.
>>
>>I'm telling you that the error with Sample.Maui within the Shiny repo that you linked in your first comment is from a provisioning problem.  That is fact - the build message says so.  I have a labelled appId in called org.shiny.samples which means your profile has to match that.  There is nothing more I can do to help you there. 
>>
>>The template also isn't giving me any errors or build time exceptions.  What is the exception you are getting?
>>
>>I am always using the latest version of visual studio and xcode
>>
>>-- 
>>Reply to this email directly or view it on GitHub:
>>https://github.com/orgs/shinyorg/discussions/1147#discussioncomment-4286636
>>You are receiving this because you authored the thread.
>>
>>Message ID: ***@***.***>
>-- 
>Sent from my Android device with K-9 Mail. Please excuse my brevity.
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Reply by aritchie** [reply]
>  I am spending a lot of time trying to get it to work, it doesnt on iOS so some form of help is needed

You are treating this like tech support.  You haven't given me anything reproducible nor a stack trace.  You are expecting me to figure this out for you.  You've been given help, but it didn't work for you.  Your issue is not my issue.  I've tried to reproduce it without any success on my mac and my windows machine with both template & sample.  I used more of my time than I should have to do it only for you to continue pestering with no additional or helpful information.... again.... ON AN ALPHA

> We would love to pay for this solutiion but, it must work out of the box and not require huge effort to get things working.

"We'll pay if you help us" is how that reads which is not something I do.  There is a stable version.  If you need the MAUI stuff, you either need to wait, learn to work with what is available, or find something else.

> To say 'it works on my machine' is the sign that it has not been tested in a new and clean installation.

The debate is done as you've used enough of my time.
